### ISSUE TITLE
Bug: Optional Rule - Hidden/Invisible Attackers automatically have Advantage gives tokens with Tremorsense advantage

### ISSUE THREAD

**Issue by tposney** [tposney/midi-qol#1162]
In GitLab by @ZBell on Feb 16, 2023, 04:09

When the optional rule, Hidden/Invisible Attacks automatically have advantage, is toggled on a token that has the Tremorsense vision type is automatically given advantage when making attacks.  The screenshot is from a different world, but I also tested this in a world with only MidiQoL and its dependencies enabled, was I missing a required module?  ![Midi-QoL-HiddenAdv-Tremorsense](https://my-gitlab-bucket.s3.amazonaws.com/640711674/7461c13995cc3282765c7486cdcd7f7700604f3aa9f1e4aef1f0e3351dc4bfe2/Midi-QoL-HiddenAdv-Tremorsense.PNG)

**Comment by tposney** [comment]
Make sure that the target can see the attacker - if they can't then the attacker gets advantage.

I just tested with a token with tremorsense and it is not automatically getting advantage.

That does not mean there is no bug, but we need to make sure that the target can see the attacker before being certain there is a problem.

**Comment by tposney** [comment]
In GitLab by @ZBell on Feb 22, 2023, 03:52

You are correct, I made an error, I saw the default Basic Vision in the token config and didn't realize it was set to 0.  Giving them proper vision results in it working as expected.

I apologize for any wasted time, thank you for all your hard work!
![image](https://my-gitlab-bucket.s3.amazonaws.com/640711674/90ccb2ea9cb6fc697e2eae3518025b536011f0e2dace9cb78d3f51c0f577ef47/image.png)
